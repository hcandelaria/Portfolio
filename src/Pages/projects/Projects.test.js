import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Projects from '.';

describe('Projects page', () => {
  const projects = [
    {
      id: '1', name: 'Data Visualization',
      description: 'Data Visuaization for my data analysis.',
      analisys: ['Python', 'Pandas', 'NumPY', 'Matplotlib'],
      backend: ['REST API', 'Node', 'Axios', 'React-Router'],
      frontend: ['React', 'HTML5', 'CSS3', 'TypeScript', 'Ionic', 'Rechart', 'Nivo'],
      links: ['https://github.com/hcandelaria/dataVisualization', 'https://data.hcandelaria.com'],
      images: [
        "https://hcandelariapersonalportfolio.s3.amazonaws.com/images/dataVisualization/dataVisualization1.png",
        "https://hcandelariapersonalportfolio.s3.amazonaws.com/images/dataVisualization/dataVisualization2.png",
        "https://hcandelariapersonalportfolio.s3.amazonaws.com/images/dataVisualization/dataVisualization3.png",
        "https://hcandelariapersonalportfolio.s3.amazonaws.com/images/dataVisualization/dataVisualization4.png",
        "https://hcandelariapersonalportfolio.s3.amazonaws.com/images/dataVisualization/dataVisualization5.png"],
      services: ['AWS S3', 'AWS Amplify', 'AWS Route53'],
      // test: ['Jest', 'Enzyme'],
    }, {
      id: '0', name: 'Data Visualization',
      description: 'Data Visuaization for my data analysis.',
      analisys: ['Python', 'Pandas', 'NumPY', 'Matplotlib'],
      backend: ['REST API', 'Node', 'Axios', 'React-Router'],
      links: ['https://github.com/hcandelaria/dataVisualization', 'https://data.hcandelaria.com'],
      frontend: ['React', 'HTML5', 'CSS3', 'TypeScript', 'Ionic', 'Rechart', 'Nivo'],
      images: [
        "https://hcandelariapersonalportfolio.s3.amazonaws.com/images/dataVisualization/dataVisualization1.png",
        "https://hcandelariapersonalportfolio.s3.amazonaws.com/images/dataVisualization/dataVisualization2.png",
        "https://hcandelariapersonalportfolio.s3.amazonaws.com/images/dataVisualization/dataVisualization3.png",
        "https://hcandelariapersonalportfolio.s3.amazonaws.com/images/dataVisualization/dataVisualization4.png",
        "https://hcandelariapersonalportfolio.s3.amazonaws.com/images/dataVisualization/dataVisualization5.png"],
      services: ['AWS S3', 'AWS Amplify', 'AWS Route53'],
      test: ['Jest', 'Enzyme'],
    }
  ]

  it('should mounts', () => {
    render(<Projects />);
    const projectsPage = screen.getByTestId('projects-page');
    expect(projectsPage).toBeInTheDocument();
  });

  it('should render a loading status', () => {
    render(<Projects />)
    const loadingEl = screen.getByRole("status")
    expect(loadingEl).toBeInTheDocument()
  })

  it('should render 2 project', () => {
    render(<Projects projects={projects} />)
    const cards = screen.getAllByRole("note")
    expect(cards.length).toEqual(2)
  })

  it('should render correctly', () => {
    const tree = renderer.create(<Projects projects={projects} />).toJSON()
    expect(tree).toMatchSnapshot();
  })

});
