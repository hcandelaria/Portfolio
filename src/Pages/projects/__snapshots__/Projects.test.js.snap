// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects page should render correctly 1`] = `
<section
  className="min-h-screen items-center justify-center py-4 sm:ml-40"
  data-testid="projects-page"
  id="projects-page"
  style={Object {}}
>
  <ul
    className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4"
    id="projects-container"
    style={
      Object {
        "opacity": 1,
        "transform": "scale(0) translateZ(0)",
      }
    }
  >
    <div
      className="mx-4 mt-5 flex justify-center bg-black-russian-700"
      role="note"
      style={
        Object {
          "opacity": 0,
          "transform": "translateY(100px) translateZ(0)",
        }
      }
    >
      <div
        className="max-w-sm rounded-lg"
      >
        <div
          className="carousel slide relative p-2"
          data-bs-ride="carousel"
          id="imageCarousel"
        >
          <div
            className="carousel-inner relative flex w-full overflow-hidden"
          >
            <div
              className="carousel-item active min-w-full"
              style={
                Object {
                  "opacity": 0,
                  "transform": "translateX(300px) translateZ(0)",
                }
              }
            >
              <img
                alt="project 0"
                src="https://hcandelariapersonalportfolio.s3.amazonaws.com/images/dataVisualization/dataVisualization1.png"
                style={
                  Object {
                    "display": "block",
                  }
                }
              />
              <img
                alt="project 0"
                src="https://hcandelariapersonalportfolio.s3.amazonaws.com/images/dataVisualization/dataVisualization2.png"
                style={
                  Object {
                    "display": "none",
                  }
                }
              />
              <img
                alt="project 0"
                src="https://hcandelariapersonalportfolio.s3.amazonaws.com/images/dataVisualization/dataVisualization3.png"
                style={
                  Object {
                    "display": "none",
                  }
                }
              />
              <img
                alt="project 0"
                src="https://hcandelariapersonalportfolio.s3.amazonaws.com/images/dataVisualization/dataVisualization4.png"
                style={
                  Object {
                    "display": "none",
                  }
                }
              />
              <img
                alt="project 0"
                src="https://hcandelariapersonalportfolio.s3.amazonaws.com/images/dataVisualization/dataVisualization5.png"
                style={
                  Object {
                    "display": "none",
                  }
                }
              />
            </div>
          </div>
          <button
            className="group absolute left-0 top-0 z-10 flex h-full cursor-pointer items-center justify-center px-4 focus:outline-none"
            data-carousel-prev={true}
            onClick={[Function]}
            type="button"
          >
            <span
              className="inline-flex h-8 w-8 items-center justify-center rounded-full bg-white/30 group-hover:bg-white/50 group-focus:outline-none group-focus:ring-4 group-focus:ring-white dark:bg-gray-800/30 dark:group-hover:bg-gray-800/60 dark:group-focus:ring-gray-800/70 sm:h-10 sm:w-10"
            >
              <iconify-icon
                height="24"
                icon="line-md:arrow-small-left"
                width="24"
              />
              <span
                className="hidden"
              >
                Previous
              </span>
            </span>
          </button>
          <button
            className="group absolute right-0 top-0 z-10 flex h-full cursor-pointer items-center justify-center px-4 focus:outline-none"
            data-carousel-next={true}
            onClick={[Function]}
            type="button"
          >
            <span
              className="inline-flex h-8 w-8 items-center justify-center rounded-full bg-white/30 group-hover:bg-white/50 group-focus:outline-none group-focus:ring-4 group-focus:ring-white dark:bg-gray-800/30 dark:group-hover:bg-gray-800/60 dark:group-focus:ring-gray-800/70 sm:h-10 sm:w-10"
            >
              <iconify-icon
                height="24"
                icon="line-md:arrow-small-right"
                width="24"
              />
              <span
                className="hidden"
              >
                Next
              </span>
            </span>
          </button>
        </div>
        <div
          className="p-6"
        >
          <h5
            className="mb-2 text-xl font-medium text-gray-400"
          >
            Data Visualization
          </h5>
          <p
            className="mb-4 text-base text-gray-300"
          >
            Data Visuaization for my data analysis.
          </p>
          <div
            className="my-1 select-none justify-center whitespace-pre-wrap text-gray-300"
          >
            Frontend:
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-info text-white border-info btn-primary text-md sm:text-sm"
            >
              React
            </div>
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-info text-white border-info btn-primary text-md sm:text-sm"
            >
              HTML5
            </div>
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-info text-white border-info btn-primary text-md sm:text-sm"
            >
              CSS3
            </div>
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-info text-white border-info btn-primary text-md sm:text-sm"
            >
              TypeScript
            </div>
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-info text-white border-info btn-primary text-md sm:text-sm"
            >
              Ionic
            </div>
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-info text-white border-info btn-primary text-md sm:text-sm"
            >
              Rechart
            </div>
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-info text-white border-info btn-primary text-md sm:text-sm"
            >
              Nivo
            </div>
          </div>
          <div
            className="my-1 select-none justify-center whitespace-pre-wrap text-gray-300"
          >
            Backend:
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-warning text-white border-warning btn-primary text-md sm:text-sm"
            >
              REST API
            </div>
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-warning text-white border-warning btn-primary text-md sm:text-sm"
            >
              Node
            </div>
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-warning text-white border-warning btn-primary text-md sm:text-sm"
            >
              Axios
            </div>
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-warning text-white border-warning btn-primary text-md sm:text-sm"
            >
              React-Router
            </div>
          </div>
          <div
            className="my-1 select-none justify-center whitespace-pre-wrap text-gray-300"
          >
            Services:
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-primary text-white border-primary btn-primary text-md sm:text-sm"
            >
              AWS S3
            </div>
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-primary text-white border-primary btn-primary text-md sm:text-sm"
            >
              AWS Amplify
            </div>
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-primary text-white border-primary btn-primary text-md sm:text-sm"
            >
              AWS Route53
            </div>
          </div>
          <div
            className="my-1 select-none justify-center whitespace-pre-wrap text-gray-300"
          >
            Analisys:
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-danger text-white border-danger btn-primary text-md sm:text-sm"
            >
              Python
            </div>
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-danger text-white border-danger btn-primary text-md sm:text-sm"
            >
              Pandas
            </div>
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-danger text-white border-danger btn-primary text-md sm:text-sm"
            >
              NumPY
            </div>
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-danger text-white border-danger btn-primary text-md sm:text-sm"
            >
              Matplotlib
            </div>
          </div>
          <a
            className="icon-link"
            href="https://github.com/hcandelaria/dataVisualization"
            rel="noreferrer"
            target="_blank"
          >
            <iconify-icon
              globe-outlineheight="32px"
              icon="line-md:github"
              width="32px"
            />
          </a>
          <a
            className="icon-link"
            href="https://data.hcandelaria.com"
            rel="noreferrer"
            target="_blank"
          >
            <iconify-icon
              globe-outlineheight="32px"
              icon="line-md:cloud"
              width="32px"
            />
          </a>
        </div>
      </div>
    </div>
    <div
      className="mx-4 mt-5 flex justify-center bg-black-russian-700"
      role="note"
      style={
        Object {
          "opacity": 0,
          "transform": "translateY(100px) translateZ(0)",
        }
      }
    >
      <div
        className="max-w-sm rounded-lg"
      >
        <div
          className="carousel slide relative p-2"
          data-bs-ride="carousel"
          id="imageCarousel"
        >
          <div
            className="carousel-inner relative flex w-full overflow-hidden"
          >
            <div
              className="carousel-item active min-w-full"
              style={
                Object {
                  "opacity": 0,
                  "transform": "translateX(300px) translateZ(0)",
                }
              }
            >
              <img
                alt="project 0"
                src="https://hcandelariapersonalportfolio.s3.amazonaws.com/images/dataVisualization/dataVisualization1.png"
                style={
                  Object {
                    "display": "block",
                  }
                }
              />
              <img
                alt="project 0"
                src="https://hcandelariapersonalportfolio.s3.amazonaws.com/images/dataVisualization/dataVisualization2.png"
                style={
                  Object {
                    "display": "none",
                  }
                }
              />
              <img
                alt="project 0"
                src="https://hcandelariapersonalportfolio.s3.amazonaws.com/images/dataVisualization/dataVisualization3.png"
                style={
                  Object {
                    "display": "none",
                  }
                }
              />
              <img
                alt="project 0"
                src="https://hcandelariapersonalportfolio.s3.amazonaws.com/images/dataVisualization/dataVisualization4.png"
                style={
                  Object {
                    "display": "none",
                  }
                }
              />
              <img
                alt="project 0"
                src="https://hcandelariapersonalportfolio.s3.amazonaws.com/images/dataVisualization/dataVisualization5.png"
                style={
                  Object {
                    "display": "none",
                  }
                }
              />
            </div>
          </div>
          <button
            className="group absolute left-0 top-0 z-10 flex h-full cursor-pointer items-center justify-center px-4 focus:outline-none"
            data-carousel-prev={true}
            onClick={[Function]}
            type="button"
          >
            <span
              className="inline-flex h-8 w-8 items-center justify-center rounded-full bg-white/30 group-hover:bg-white/50 group-focus:outline-none group-focus:ring-4 group-focus:ring-white dark:bg-gray-800/30 dark:group-hover:bg-gray-800/60 dark:group-focus:ring-gray-800/70 sm:h-10 sm:w-10"
            >
              <iconify-icon
                height="24"
                icon="line-md:arrow-small-left"
                width="24"
              />
              <span
                className="hidden"
              >
                Previous
              </span>
            </span>
          </button>
          <button
            className="group absolute right-0 top-0 z-10 flex h-full cursor-pointer items-center justify-center px-4 focus:outline-none"
            data-carousel-next={true}
            onClick={[Function]}
            type="button"
          >
            <span
              className="inline-flex h-8 w-8 items-center justify-center rounded-full bg-white/30 group-hover:bg-white/50 group-focus:outline-none group-focus:ring-4 group-focus:ring-white dark:bg-gray-800/30 dark:group-hover:bg-gray-800/60 dark:group-focus:ring-gray-800/70 sm:h-10 sm:w-10"
            >
              <iconify-icon
                height="24"
                icon="line-md:arrow-small-right"
                width="24"
              />
              <span
                className="hidden"
              >
                Next
              </span>
            </span>
          </button>
        </div>
        <div
          className="p-6"
        >
          <h5
            className="mb-2 text-xl font-medium text-gray-400"
          >
            Data Visualization
          </h5>
          <p
            className="mb-4 text-base text-gray-300"
          >
            Data Visuaization for my data analysis.
          </p>
          <div
            className="my-1 select-none justify-center whitespace-pre-wrap text-gray-300"
          >
            Frontend:
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-info text-white border-info btn-primary text-md sm:text-sm"
            >
              React
            </div>
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-info text-white border-info btn-primary text-md sm:text-sm"
            >
              HTML5
            </div>
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-info text-white border-info btn-primary text-md sm:text-sm"
            >
              CSS3
            </div>
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-info text-white border-info btn-primary text-md sm:text-sm"
            >
              TypeScript
            </div>
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-info text-white border-info btn-primary text-md sm:text-sm"
            >
              Ionic
            </div>
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-info text-white border-info btn-primary text-md sm:text-sm"
            >
              Rechart
            </div>
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-info text-white border-info btn-primary text-md sm:text-sm"
            >
              Nivo
            </div>
          </div>
          <div
            className="my-1 select-none justify-center whitespace-pre-wrap text-gray-300"
          >
            Backend:
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-warning text-white border-warning btn-primary text-md sm:text-sm"
            >
              REST API
            </div>
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-warning text-white border-warning btn-primary text-md sm:text-sm"
            >
              Node
            </div>
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-warning text-white border-warning btn-primary text-md sm:text-sm"
            >
              Axios
            </div>
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-warning text-white border-warning btn-primary text-md sm:text-sm"
            >
              React-Router
            </div>
          </div>
          <div
            className="my-1 select-none justify-center whitespace-pre-wrap text-gray-300"
          >
            Services:
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-primary text-white border-primary btn-primary text-md sm:text-sm"
            >
              AWS S3
            </div>
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-primary text-white border-primary btn-primary text-md sm:text-sm"
            >
              AWS Amplify
            </div>
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-primary text-white border-primary btn-primary text-md sm:text-sm"
            >
              AWS Route53
            </div>
          </div>
          <div
            className="my-1 select-none justify-center whitespace-pre-wrap text-gray-300"
          >
            Analisys:
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-danger text-white border-danger btn-primary text-md sm:text-sm"
            >
              Python
            </div>
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-danger text-white border-danger btn-primary text-md sm:text-sm"
            >
              Pandas
            </div>
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-danger text-white border-danger btn-primary text-md sm:text-sm"
            >
              NumPY
            </div>
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-danger text-white border-danger btn-primary text-md sm:text-sm"
            >
              Matplotlib
            </div>
          </div>
          <div
            className="my-1 select-none justify-center whitespace-pre-wrap text-gray-300"
          >
            Test:
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-success text-white border-success btn-primary text-md sm:text-sm"
            >
              Jest
            </div>
            <div
              className="px-1 m-1 inline-block shadow-md no-underline rounded-full bg-success text-white border-success btn-primary text-md sm:text-sm"
            >
              Enzyme
            </div>
          </div>
          <a
            className="icon-link"
            href="https://github.com/hcandelaria/dataVisualization"
            rel="noreferrer"
            target="_blank"
          >
            <iconify-icon
              globe-outlineheight="32px"
              icon="line-md:github"
              width="32px"
            />
          </a>
          <a
            className="icon-link"
            href="https://data.hcandelaria.com"
            rel="noreferrer"
            target="_blank"
          >
            <iconify-icon
              globe-outlineheight="32px"
              icon="line-md:cloud"
              width="32px"
            />
          </a>
        </div>
      </div>
    </div>
  </ul>
</section>
`;
